go/logic/applier.go

```diff
@@ -53,12 +53,14 @@ func (this *Applier) InitDBConnections() (err error) {
 		return err
 	}
 	this.singletonDB.SetMaxOpenConns(1)
-	if err := this.validateConnection(this.db); err != nil {
+	version, err := base.ValidateConnection(this.db, this.connectionConfig)
+	if err != nil {
 		return err
 	}
-	if err := this.validateConnection(this.singletonDB); err != nil {
+	if _, err := base.ValidateConnection(this.singletonDB, this.connectionConfig); err != nil {
```

Comment:
This exposed kind of a problem with the original code. We would overwrite the ApplierMySQLVersion with this second call. I almost wrote it that way, but I wasn't entirely sure about `this.db` vs `this.singletonDB`. So let know how you want me to handle it.

I chose the first connection `this.db` to set the version.

Reply:
I'll look into it.